iceoryx2-ffi/cxx/include/iox2/service_builder_event.hpp

```diff
@@ -48,6 +49,30 @@ class ServiceBuilderEvent {
     /// [`Listener`] must be at least supported.
     IOX_BUILDER_OPTIONAL(uint64_t, max_listeners);
 
+    /// If the [`Service`] is created it defines the event that shall be emitted by every newly
+    /// created [`Notifier`].
+    IOX_BUILDER_OPTIONAL(EventId, notifier_created_event);
+
+    /// If the [`Service`] is created it disables sending an event when a new notifier was created.
+    IOX_BUILDER_SWITCH(disable_notifier_created_event);
+
+    /// If the [`Service`] is created it defines the event that shall be emitted by every
+    /// [`Notifier`] before it is dropped. If [`None`] is
+    /// provided a [`Notifier`] will not emit an event.
+    IOX_BUILDER_OPTIONAL(EventId, notifier_dropped_event);
+
+    /// If the [`Service`] is created it disables sending an event when a notifier was dropped.
+    IOX_BUILDER_SWITCH(disable_notifier_dropped_event);
+
+    /// If the [`Service`] is created it defines the event that shall be emitted when a
+    /// [`Notifier`] is identified as dead. If [`None`] is
+    /// provided no event will be emitted.
+    IOX_BUILDER_OPTIONAL(EventId, notifier_dead_event);
+
+    /// If the [`Service`] is created it disables sending an event when a notifier was identified
+    /// as dead.
+    IOX_BUILDER_SWITCH(disable_notifier_dead_event);
```

Comment:
```suggestion
    /// If the [`Service`] is created it defines the event that shall be emitted by every newly
    /// created [`Notifier`].
    IOX_BUILDER_OPTIONAL(EventId, notifier_created_event);

    /// If the [`Service`] is created it defines the event that shall be emitted by every
    /// [`Notifier`] before it is dropped. If [`None`] is
    /// provided a [`Notifier`] will not emit an event.
    IOX_BUILDER_OPTIONAL(EventId, notifier_dropped_event);

    /// If the [`Service`] is created it defines the event that shall be emitted when a
    /// [`Notifier`] is identified as dead. If [`None`] is
    /// provided no event will be emitted.
    IOX_BUILDER_OPTIONAL(EventId, notifier_dead_event);
    
    /// If the [`Service`] is created it disables sending an event when a new notifier was created.
    void disable_notifier_created_event() {
        m_notifier_created_event.reset();
    }
    
    /// If the [`Service`] is created it disables sending an event when a notifier was dropped.
    void disable_notifier_dropped_event() {
        m_notifier_dropped_event.reset();
    }
    
    /// If the [`Service`] is created it disables sending an event when a notifier was identified
    /// as dead.
    void disable_notifier_dead_event() {
        m_notifier_dead_event.reset();
    }
```

No need to make it unnecessary complicated. This will also simplify the logic in `service_builder_event.cpp`

Reply:
Actually, this does not work. When this is not called, the user has not set the requirement and takes whatever QoS the service is offering. As soon as you call disable or enable with a specific event, you set this as a requirement that is ensured when opening the service.

Reply:
Okay, then I do not understand what you want to achieve :sweat_smile:

Reply:
Hmm, if it's three options, a new enum might indeed make sense
```rs
enum Event {
    Default, // can be on or off; maybe Preconfigured
    Custom{id},
    Disabled,
}
```

Reply:
We can go that way, but then we need this for every builder option in every factory for all ports and services since it is always handled like this.

I think your suggestion is not bad at all, but would require a greater refactoring. My suggestion for now:
1. Pursue the current approach and merge the PR
2. Create an issue to track this and tackle this in v0.5 or later